pkg/kubelet/prober/worker.go

```diff
@@ -314,6 +322,29 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 		return true
 	}
 
+	if resultChanged {
```

Comment:
if `resultChanged` is `true`, `w.resultRun` is reset to `1` at L316. Unless `w.spec.*Threshold` is `1`, this function finishes at L322. If the threshold in the spec is larger than `1`, the added events will be never emitted. Is this the expected behavior?

Reply:
I think this should be:
```suggestion
	if (result == results.Failure && w.resultRun == int(w.spec.FailureThreshold)) ||
		(result == results.Success && w.resultRun == int(w.spec.SuccessThreshold)) {
```
Then, an event will be emitted only when a counter has reached a threshold.